### ISSUE TITLE
Connection dialog does not show script icon in advanced mode

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot#92513]
### Tested versions

4.3 dev6 and earlier

### System information

Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1060 (NVIDIA; 31.0.15.4633) - Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz (8 Threads)

### Issue description

https://github.com/godotengine/godot/assets/2223172/c0b3c703-f787-4c9b-a3a1-121983be2c99

This is caused by `connect_to_script_mode`, which either shows scripts, but blocks other nodes, or allows all nodes, but hides scripts. I don't see a reason why the scripts should be hidden, they provide more context when looking at the scene tree.

### Steps to reproduce

1. Have a scene with scripts
2. Go to Node -> Signals and double-click a signal
3. Toggle Advanced and compare the scene view

### Minimal reproduction project (MRP)

N/A

**Comment by stephen-berry** [comment]
After a quick glance at the code, it seemed like there was purposeful intention to hide the script icon. Namely, advanced mode being enabled swaps the usual "Connect to Script" text to "Connect to Mode" (which as you saw removes the script).

Class: **connections_dialog** Method: **ConnectDialog::_advanced_pressed()**
```

void ConnectDialog::_advanced_pressed() {
	if (advanced->is_pressed()) {
		connect_to_label->set_text(TTR("Connect to Node:"));
		tree->set_connect_to_script_mode(false); <-----
		vbc_right->show();
		error_label->hide();
	} else {
		reset_size();
		connect_to_label->set_text(TTR("Connect to Script:"));
		tree->set_connect_to_script_mode(_true_); <-----
		vbc_right->hide();
		error_label->set_visible(!_find_first_script(get_tree()->get_edited_scene_root(), get_tree()->get_edited_scene_root()));
	}
	EditorSettings::get_singleton()->set_project_metadata("editor_metadata", "use_advanced_connections", advanced->is_pressed());
	popup_centered();
}
```

Usually, I'd assume this means it's an improvement over a bug. However, advanced mode doesn't suddenly allow users to connect a signal to a node without a script (meaning nothing changes in terms of which nodes can actually be selected). Unless someone thinks otherwise, I would assume the initial intentions are more confusing than necessary.

I can upload the change which would involve keeping the "Connect to Script" text regardless, and won't disable the connect_to_script_mode(). If anyone notices where I may be missing something, let me know :)

**Comment by KoBeWi** [comment]
> advanced mode doesn't suddenly allow users to connect a signal to a node without a script

It does. In regular mode, scriptless nodes are unselectable, in advanced mode you can select any node and pick a non-script method.

I'm not sure if fix is trivial, because SceneTreeEditor is used not only in the connection dialog and the changes shouldn't affect other trees.

**Comment by stephen-berry** [comment]
Thanks for the heads up! I didn't notice that while playing around (let alone non-script methods!). I'd like to have a deeper look to ensure showing the icons here don't affect other areas of code.

**Comment by akien-mga** [comment]
Fixed by #92648.
